components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/Logins.kt

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.sync.logins
+
+import mozilla.components.concept.storage.*
+
+/// Convert between application-services data classes and the ones in
+/// concepts.storage.
+
+fun mozilla.appservices.logins.Login.toAC() = Login(
+    guid = record.id,
+    origin = fields.origin,
+    formActionOrigin = fields.formActionOrigin,
+    httpRealm = fields.httpRealm,
+    usernameField = fields.usernameField,
+    passwordField = fields.passwordField,
+    timesUsed = record.timesUsed,
+    timeCreated = record.timeCreated,
+    timeLastUsed = record.timeLastUsed,
+    timePasswordChanged = record.timePasswordChanged,
+    username = secFields.username,
+    password = secFields.password,
+)
+
+fun mozilla.appservices.logins.EncryptedLogin.toAC() = EncryptedLogin(
+    guid = record.id,
+    origin = fields.origin,
+    formActionOrigin = fields.formActionOrigin,
+    httpRealm = fields.httpRealm,
+    usernameField = fields.usernameField,
+    passwordField = fields.passwordField,
+    timesUsed = record.timesUsed,
+    timeCreated = record.timeCreated,
+    timeLastUsed = record.timeLastUsed,
+    timePasswordChanged = record.timePasswordChanged,
+    secureFields = secFields,
```

Comment:
I think part of it was the extra decryption step and another part was how to manage the data.  If you decrypt the username, how do you store the decrypted username with the login so you don't need to keep decrypting it?  I think the alternative would be to have a `Login` type where each piece of data was sometimes encrypted and sometimes decrypted.  But that can get confusing quickly.  If you have a list of logins, how do you know which data is decrypted or not?

It seemed simpler to do all the decryption in one pass and the goal would be that that pass would happen when the user clicks on "suggested logins" and  we needed to display the login list to show the user.  This is also why `decryptLogins` works on a list rather than an individual login (of course if you wanted a single-item function, we could definitely add that).

Reply:
BTW, I also would personally push for storing the username plaintext.  I don't really feel like it's sensitive data in general.  But I think that would require a larger discussion.